### PR TITLE
Filtering: remove invisible characters before checking filters

### LINKED ISSUE #1469: Invisible characters can be used to bypass filters
Using zero width spaces and similar invisible characters, it is possible to easily bypass our token filter list.

We should strip such characters before running the regexes.

### PR DESCRIPTION
**Pull request by MarkKoz**
Fixes #1469 

Invisible characters are removed for the following filters:

* names
* urls
* invites
* regex

### CODE REVIEW

**Review by Akarys42 — APPROVED** [review]
Lgtm, thanks!

**Review by ChrisLovering — APPROVED** [review]
Looks good!

File: bot/exts/filters/filtering.py [hunk 1]
```diff
@@ -628,6 +634,22 @@ async def delete_offensive_msg(self, msg: Mapping[str, str]) -> None:
...
+        Removed characters:
+
+        - mongolian vowel separator
+        - zero width space
+        - zero width non-joiner
+        - zero width joiner
+        - word joiner
+        - zero width non-breaking space
+        """
+        return re.sub("[\u180e\u200b\u200c\u200d\u2060\ufeff]", "", string)
```
Comment (Numerlor) [inline bot/exts/filters/filtering.py:651]: Could avoid the docstring commenting on the character names if named escapes would be used here

**Review by Numerlor — COMMENTED** [review]
Do we only want to handle the common chars  here?

There's a bunch more of chars which don't render to anything noticeable, I haven't checked them all for how they look on discord but the tags for example don't show up as anything https://paste.fuelrats.com/oguvesutip.py

**Comment by MarkKoz** [comment]
> 
> 
> Do we only want to handle the common chars here?
> 
> There's a bunch more of chars which don't render to anything noticeable, I haven't checked them all for how they look on discord but the tags for example don't show up as anything https://paste.fuelrats.com/oguvesutip.py

I'll try to make it more thorough. Thanks for the tip about the Unicode category. Only some Arabic characters at the start render, but I think it's fine to strip them out anyway cause it seems more likely that they'd be used to circumvent filters rather than for their intended purposes considering

1. This is an English community
2. It wouldn't make sense to use Arabic characters right next to English ones.

File: bot/exts/filters/filtering.py [hunk 2]
```diff
@@ -34,7 +35,11 @@
 EVERYONE_PING_RE = re.compile(rf"@everyone|<@&{Guild.id}>|@here")
 SPOILER_RE = re.compile(r"(\|\|.+?\|\|)", re.DOTALL)
 URL_RE = re.compile(r"(https?://[^\s]+)", flags=re.IGNORECASE)
-ZALGO_RE = re.compile(r"[\u0300-\u036F\u0489]")
+
+# Exclude variation selectors from zalgo because they're actually invisible.
+VARIATION_SELECTORS = r"\uFE00-\uFE0F\U000E0100-\U000E01EF"
+INVISIBLE_RE = regex.compile(rf"[{VARIATION_SELECTORS}\p{{Cn}}\p{{FORMAT}}\p{{CONTROL}}--\s]", regex.V1)
```
[the remark this answers is not in the document]
Comment (MarkKoz) [reply bot/exts/filters/filtering.py]: Thanks, I fixed this in the rebase (I needed to fix the commit message anyway).

**Review by Numerlor — APPROVED** [review]

**Review by ChrisLovering — APPROVED** [review]
Tested the new regex locally, looks good to me

**Pull request merged**